src/OrchardCore.Modules/OrchardCore.Settings/Recipes/SettingsStep.cs

```diff
@@ -48,7 +60,10 @@ protected override async Task HandleAsync(RecipeExecutionContext context)
                     break;
 
                 case "ResourceDebugMode":
-                    site.ResourceDebugMode = (ResourceDebugMode)property.Value.Value<int>();
+                    if (property.Value.TryGetEnumValue<ResourceDebugMode>(out var resourceDebugMode))
```

Comment:
Why did you not do the same check everywhere else? Like in the CacheMode and MaxPagedCount, etc? 

I suggest re-adding the TryGetValue as it'll make it safer